### PR TITLE
[Color] Add UIColor+MaterialDynamic to support dynamic color.

### PR DESCRIPTION
**Pull request by fluxminds**
DynamicColor is not only useful inside color scheme class, but also in some other classes (dragon controller) as a utility method. A new UIColor category is created in this PR to facilitate it.

Cleaning up in color scheme will be done in a separate PR.

### CODE REVIEW

**Review by bryanoltman — APPROVED** [review]

File: components/private/Color/src/UIColor+MaterialDynamic.m [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "UIColor+MaterialDynamic.h"
+
+@implementation UIColor (MaterialDynamic)
+
++ (UIColor *)colorWithColorForUserInterfaceStyleDark:(UIColor *)darkColor
```
Comment (yarneo) [inline components/private/Color/src/UIColor+MaterialDynamic.m]: maybe:
`colorWithUserInterfaceStyleDarkColor:defaultColor:` ?

File: catalog/MDCDragons.xcodeproj/xcshareddata/xcschemes/MDCDragons.xcscheme [hunk 2]
```diff
@@ -28,7 +28,7 @@
             buildForAnalyzing = "NO">
```
Comment (yarneo) [inline catalog/MDCDragons.xcodeproj/xcshareddata/xcschemes/MDCDragons.xcscheme:28]: is the changes in this file needed?

File: components/private/Color/src/UIColor+MaterialDynamic.m (hunk 1 above)
Comment (fluxminds) [reply components/private/Color/src/UIColor+MaterialDynamic.m]: Sure. Addressed in 058fb4c168b60757d6b49f5c7ec539f372700602.

File: catalog/MDCDragons.xcodeproj/xcshareddata/xcschemes/MDCDragons.xcscheme (hunk 2 above)
Comment (fluxminds) [reply catalog/MDCDragons.xcodeproj/xcshareddata/xcschemes/MDCDragons.xcscheme:28]: It is needed for adding test target into scheme.

**Review by yarneo — APPROVED** [review]

**Pull request merged**